node/builder.go

```diff
@@ -648,3 +648,84 @@ func Test() Option {
 		Override(new(beacon.Schedule), testing.RandomBeacon),
 	)
 }
+
+// --------------------------------------------
+// FIXME: Move to a separate `builder` package.
+// --------------------------------------------
+
+type Builder struct {
+	// FIXME: General profile, maybe following node type for now, should
+	//  be decoupled later.
+	// Full, lite, miner.
+
+	// Configuration options that will affect the DI graph, applied at the end.
+	// FIXME: Consider making configuration options private and exposing functions
+	//  to set them. For now it doesn't seem worth it as they are mostly constants
+	//  that we don't process (maybe the ShutdownChan/Repo are the exception).
```

Comment:
We should try to design this well on the first try, things like this tend to be really painful to change.

I feel like functional options for this would work, but get a bit weird with multiple levels of them (we already have 2 levels, builder.Option and fx.Option, this would be adding a third level) - though maybe it can work nicely

Reply:
Will think about this some more. I think adding functional options for every setting will result on even more boilerplate. Most are constants with a sane default as `nil` value if left uninitialized. Maybe for the more complex ones.

Reply:
@magik6k Looking for more feedback on new commit https://github.com/filecoin-project/lotus/pull/5392/commits/0b67f0e02d0f322b232ed4b4b7a125ce8602e3eb. I still prefer to avoid adding yet another layer of functional options. I think there is a list of options we can safely expose with sane default values and abstract away other more complex ones that may need some pre/post-processing (in the example commit here the `repo` configuration option). I don't strongly advocate for this way, open to iterating on it a some more.

Reply:
Moved settings to separate config struct https://github.com/filecoin-project/lotus/pull/5392/commits/bf2d40c19978e5993f3bd8261d41b363195e449a.